specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1182 @@
+        ProgressBar UseValueStringOverride();
+
+        // Retrieves the XML content of the ProgressBar
+        String GetXml();
+    };
+
+    enum MSWinSoundEvent
+    {
+        Default,
+        IM,
+        Mail,
+        Reminder,
+        SMS,
+        Alarm,
+        Alarm2,
+        Alarm3,
+        Alarm4,
+        Alarm5,
+        Alarm6,
+        Alarm7,
+        Alarm8,
+        Alarm9,
+        Alarm10,
+        Call,
+        Call2,
+        Call3,
+        Call4,
+        Call5,
+        Call6,
+        Call7,
+        Call8,
+        Call9,
+        Call10,
+    };
+
+    enum ScenarioType
+    {
+        Default, // The normal AppNotification behavior. The AppNotification appears for a short duration, and then automatically dismisses into Notification Center.
+        Reminder, // The notification will stay on screen until the user dismisses it or takes action.
+        Alarm,    // Alarms behave like Reminder, but alarms will additionally loop audio with a default alarm sound.
+        IncomingCall, // Incoming call notifications are displayed pre-expanded in a special call format and stay on the user's screen till dismissed.
+        Urgent, // Important notifications allow users to have more control over what 1st party and 3rd party apps can send them high-priority AppNotifications (urgent/important) that can break through Focus Assist.
+    };
+
+    enum Duration
+    {
+        Short, // Default value. AppNotification appears for a short while and then goes into Notification Center.
+        Long, // AppNotification stays on-screen for longer, and then goes into Notification Center.
+    };
+
+    runtimeclass AppNotificationContent
+    {
+        AppNotificationContent();
+
+        // Adds arguments to the launch attribute to return when AppNotification is clicked.
+        AppNotificationContent AddArgument(String key, String value);
+
+        // Sets the timeStamp of the AppNotification to when it was constructed instead of when it was sent.
+        AppNotificationContent SetTimeStamp(Windows.Foundation.DateTime timeStamp);
+
+        // Sets the scenario of the AppNotification.
+        AppNotificationContent SetScenarioType(ScenarioType scenarioType);
+
+        // Adds text to the AppNotification.
+        AppNotificationContent AddText(String text);
+        AppNotificationContent AddText(String text, TextProperties properties);
+
+        AppNotificationContent SetAttributionText(String text);
+        AppNotificationContent SetAttributionText(String text, String language);
+
+        // Sets the full-width inline-image that appears when you expand the AppNotification
+        AppNotificationContent SetInlineImage(Windows.Foundation.Uri imageUri);
+        [default_overload]
+        AppNotificationContent SetInlineImage(Windows.Foundation.Uri imageUri, String alternateText);
+        AppNotificationContent SetInlineImage(Windows.Foundation.Uri imageUri, Boolean useCircleCrop);
```

Comment:
I think an enum would be preferable to a bool for useCircleCrop, no? I hate seeing a true / false argument in a function call as it doesn't tell what it is used for. Also an enum would help future proof this, in case other crop options become available in the future.

@andrewleader - What do you think?

Reply:
I'm going to change it to SetInlineImageWithCircleCrop.